kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -315,6 +325,37 @@ public String createConsumer(String group, ConsumerInstanceConfig instanceConfig
     executor.submit(new RunnableReadTask(new ReadTaskState(task, state, callback)));
   }
 
+  class ReadFutureTask<V> extends FutureTask<V> {
+
+    private Runnable myTask;
+
+    public ReadFutureTask(Runnable runnable, V result) {
+      super(runnable, result);
+      this.myTask = runnable;
+    }
+
+    public Runnable getReadTask() {
+      return myTask;
+    }
+  }
+
+  class KafkaConsumerThreadPoolExecutor extends ThreadPoolExecutor {
+    public KafkaConsumerThreadPoolExecutor(int corePoolSize,
+                                               int maximumPoolSize,
+                                               long keepAliveTime,
+                                               TimeUnit unit,
+                                               BlockingQueue<Runnable> workQueue,
+                                               RejectedExecutionHandler handler) {
+    super(corePoolSize, maximumPoolSize, keepAliveTime, unit, workQueue, handler);
+
+    }
+
+    @Override
+    protected <T> RunnableFuture<T> newTaskFor(Runnable runnable, T value) {
```

Comment:
Do we want to also override the `protected <T> RunnableFuture<T> newTaskFor(Callable<T> callable)` ? Which one gets called?
It might be better to override both in my opinion

Reply:
because we need to support executor.submit(runnable) or executor.submit(callable),so i override the protected <T> RunnableFuture<T> newTaskFor(Callable<T> callable) and in 
ReadFutureTask's getTask method,use the consistent code to return RunnableReadTask